middleware/middleware.go

```diff
@@ -0,0 +1,25 @@
+package middleware
+
+import (
+	"net/http"
+	"video-streaming-server/utils"
+)
+
+func AuthRequired(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		cookie, err := r.Cookie("auth_token")
+		if err != nil {
+			http.Redirect(w, r, "/login", http.StatusSeeOther)
+			return
+		}
+
+		tokenString := cookie.Value
+		token, err := utils.VerifyToken(tokenString)
```

Comment:
this verifytoken function seems incomplete in logic. please check.

Reply:
This needs a deeper study into dependency injections and we might also require looking into caching mechanisms.
Will need to talk to someone who has written a lot more Golang.

For later.